### DISCUSSION TITLE
Trouble running hover_swarm.launch

### DISCUSSION THREAD

**Discussion by tbaranoski** [USC-ACTLab/crazyswarm#421] [Q&A]
I have the Crazyflie 2.1 with Crazyradio PA. I have flashed the newest firmware on the Crazyflie (I originally tried with the older firmware outlined in crazyswarm.readthedocs.io and had no success). 
I am using the Ubuntu 20.04 VM with ROS Noetic (full version). 
I do not have a motion capture system so I set motion_capture_type: "none" in  hover_swarm.launch. 
I also changed js0 to js2 to match with my Xbox one controller in hover_swarm.launch. 
I have changed enable_logging: False to enable_logging: True. (M4 LED would only stay lit for 3 seconds when enable_logging was set to False when running hover_swarm)
I have even changed the permissions for the controller following this guide: http://wiki.ros.org/joy/Tutorials/ConfiguringALinuxJoystick

My crazyswarm workspace builds fine, however when I try:  
roslaunch crazyswarm hover_swarm.launch 
nothing happens. The Crazyflie appears to be connected to the Crazyradio (M4 light is lit red/green) but the Crazyflie does not move and the Xbox One Controller will not work with it. I get no errors from the terminal. I am hoping to get some help troubleshooting this.
The terminal output is shown below:

![image](https://user-images.githubusercontent.com/84639789/125855790-dc3e7339-91ea-4725-b2ff-765e364a8ded.png)

bitcraze@ubuntu:~/crazyswarm$ roslaunch crazyswarm hover_swarm.launch
... logging to /home/bitcraze/.ros/log/921008fc-e5ad-11eb-942f-4bcd6a5f9695/roslaunch-ubuntu-27698.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://ubuntu:38763/

SUMMARY
========

PARAMETERS
 * /crazyflieTypes/CF21SingleMarker/batteryVoltageWarning: 3.8
 * /crazyflieTypes/CF21SingleMarker/batteryVoltateCritical: 3.7
 * /crazyflieTypes/CF21SingleMarker/bigQuad: False
 * /crazyflieTypes/CF21SingleMarker/dynamicsConfiguration: 0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/mNGyro_rollpitch: 0.1
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/mNGyro_yaw: 0.1
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/pNAcc_xy: 1.0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/pNAcc_z: 2.0
 * /crazyflieTypes/CF21SingleMarker/markerConfiguration: 3
 * /crazyflieTypes/default/batteryVoltageWarning: 3.8
 * /crazyflieTypes/default/batteryVoltateCritical: 3.7
 * /crazyflieTypes/default/bigQuad: False
 * /crazyflieTypes/default/dynamicsConfiguration: 0
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/default/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/default/markerConfiguration: 0
 * /crazyflieTypes/defaultSingleMarker/batteryVoltageWarning: 3.8
 * /crazyflieTypes/defaultSingleMarker/batteryVoltateCritical: 3.7
 * /crazyflieTypes/defaultSingleMarker/bigQuad: False
 * /crazyflieTypes/defaultSingleMarker/dynamicsConfiguration: 0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/mNGyro_rollpitch: 0.2
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/mNGyro_yaw: 0.1
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/pNAcc_xy: 4.0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/pNAcc_z: 8.0
 * /crazyflieTypes/defaultSingleMarker/markerConfiguration: 3
 * /crazyflieTypes/large/batteryVoltageWarning: 11.4
 * /crazyflieTypes/large/batteryVoltateCritical: 11.1
 * /crazyflieTypes/large/bigQuad: True
 * /crazyflieTypes/large/dynamicsConfiguration: 0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_z: 4.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_xy: 30000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_z: 40000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_omega_rp: 100
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_xy: 3.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_z: 3.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_xy: 1.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_z: 2.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_xy: 8.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_z: 8.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_xy: 9000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_z: 10000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/mass: 0.38
 * /crazyflieTypes/large/firmwareParams/ctrlMel/massThrust: 5400
 * /crazyflieTypes/large/markerConfiguration: 2
 * /crazyflieTypes/medium/batteryVoltageWarning: 7.6
 * /crazyflieTypes/medium/batteryVoltateCritical: 7.4
 * /crazyflieTypes/medium/bigQuad: True
 * /crazyflieTypes/medium/dynamicsConfiguration: 0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_z: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_xy: 50000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_xy: 1.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_z: 1.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_xy: 0.1
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_z: 0.5
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_xy: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_z: 3
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_xy: 16000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/mass: 0.11
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/massThrust: 23000
 * /crazyflieTypes/medium/markerConfiguration: 1
 * /crazyflies: [{'channel': 100,...
 * /crazyswarm_server/enable_logging: True
 * /crazyswarm_server/enable_parameters: True
 * /crazyswarm_server/firmwareParams/commander/enHighLevel: 1
 * /crazyswarm_server/firmwareParams/kalman/resetEstimation: 1
 * /crazyswarm_server/firmwareParams/locSrv/extPosStdDev: 1e-3
 * /crazyswarm_server/firmwareParams/locSrv/extQuatStdDev: 0.05
 * /crazyswarm_server/firmwareParams/ring/effect: 16
 * /crazyswarm_server/firmwareParams/ring/headlightEnable: 0
 * /crazyswarm_server/firmwareParams/ring/solidBlue: 255
 * /crazyswarm_server/firmwareParams/ring/solidGreen: 0
 * /crazyswarm_server/firmwareParams/ring/solidRed: 0
 * /crazyswarm_server/firmwareParams/stabilizer/controller: 2
 * /crazyswarm_server/firmwareParams/stabilizer/estimator: 2
 * /crazyswarm_server/force_no_cache: False
 * /crazyswarm_server/genericLogTopicFrequencies: [10]
 * /crazyswarm_server/genericLogTopic_log1_Variables: ['stateEstimate.x...
 * /crazyswarm_server/genericLogTopics: ['log1']
 * /crazyswarm_server/motion_capture_type: none
 * /crazyswarm_server/object_tracking_type: libobjecttracker
 * /crazyswarm_server/print_latency: False
 * /crazyswarm_server/save_point_clouds: ~/pointCloud.ot
 * /crazyswarm_server/send_position_only: False
 * /crazyswarm_server/vicon_host_name: vicon
 * /crazyswarm_server/world_frame: /world
 * /crazyswarm_server/write_csvs: False
 * /crazyswarm_teleop/csv_file: /home/bitcraze/cr...
 * /crazyswarm_teleop/timescale: 0.8
 * /dynamicsConfigurations/0/maxFitnessScore: 0.001
 * /dynamicsConfigurations/0/maxPitch: 1.4
 * /dynamicsConfigurations/0/maxPitchRate: 20.0
 * /dynamicsConfigurations/0/maxRoll: 1.4
 * /dynamicsConfigurations/0/maxRollRate: 20.0
 * /dynamicsConfigurations/0/maxXVelocity: 2.0
 * /dynamicsConfigurations/0/maxYVelocity: 2.0
 * /dynamicsConfigurations/0/maxYawRate: 10.0
 * /dynamicsConfigurations/0/maxZVelocity: 3.0
 * /joy/dev: /dev/input/js2
 * /markerConfigurations/0/numPoints: 4
 * /markerConfigurations/0/offset: [0.0, -0.01, -0.04]
 * /markerConfigurations/0/points/0: [0.0177184, 0.013...
 * /markerConfigurations/0/points/1: [-0.0262914, 0.05...
 * /markerConfigurations/0/points/2: [-0.0328889, -0.0...
 * /markerConfigurations/0/points/3: [0.0431307, -0.03...
 * /markerConfigurations/1/numPoints: 4
 * /markerConfigurations/1/offset: [0.0, 0.0, -0.03]
 * /markerConfigurations/1/points/0: [-0.00896228, -0....
 * /markerConfigurations/1/points/1: [-0.0156318, 0.09...
 * /markerConfigurations/1/points/2: [0.0461693, -0.08...
 * /markerConfigurations/1/points/3: [-0.0789959, -0.0...
 * /markerConfigurations/2/numPoints: 4
 * /markerConfigurations/2/offset: [0.0, 0.0, -0.06]
 * /markerConfigurations/2/points/0: [0.0558163, -0.00...
 * /markerConfigurations/2/points/1: [-0.0113941, 0.00...
 * /markerConfigurations/2/points/2: [-0.0306277, 0.05...
 * /markerConfigurations/2/points/3: [0.0535816, -0.04...
 * /markerConfigurations/3/numPoints: 1
 * /markerConfigurations/3/offset: [0.0, -0.01, -0.04]
 * /markerConfigurations/3/points/0: [0.0177184, 0.013...
 * /numDynamicsConfigurations: 1
 * /numMarkerConfigurations: 4
 * /rosdistro: noetic
 * /rosversion: 1.15.11

NODES
  /
    crazyswarm_server (crazyswarm/crazyswarm_server)
    crazyswarm_teleop (crazyswarm/crazyswarm_teleop)
    joy (joy/joy_node)
    rviz (rviz/rviz)

auto-starting new master
process[master]: started with pid [27711]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 921008fc-e5ad-11eb-942f-4bcd6a5f9695
process[rosout-1]: started with pid [27726]
started core service [/rosout]
process[crazyswarm_server-2]: started with pid [27733]
process[joy-3]: started with pid [27734]
process[crazyswarm_teleop-4]: started with pid [27735]
process[rviz-5]: started with pid [27736]
[ INFO] [1626381922.546496953]: Wait for services...
[ WARN] [1626381922.547206370]: Couldn't set gain on joystick force feedback: Bad file descriptor
[ INFO] [1626381922.547745585]: waitForService: Service [/emergency] has not been advertised, waiting...
[ INFO] [1626381922.548160505]: Opened joystick: /dev/input/js2 (Microsoft X-Box One pad). deadzone_: 0.050000.
[ INFO] [1626381922.568947547]: waitForService: Service [/emergency] is now available.
[ INFO] [1626381922.570919483]: Manager ready.
ch: 1
[ INFO] [1626381922.712153577]: Adding CF: cf1 (radio://0/100/2M/E7E7E7E701, cf1)...
[ INFO] [1626381922.755996518]: CF ctor: 0.042615 s
[ INFO] [1626381922.787500029]: [cf1] SYS: ----------------------------
[ INFO] [1626381922.808319083]: [cf1] SYS: Crazyflie 2.1 is up and running!
[ INFO] [1626381922.830557199]: [cf1] SYS: Build 11:0864ef92245a (2021.03 +11) CLEAN
[ INFO] [1626381922.862602372]: [cf1] SYS: I am 0x20383347344D50080030004D and I have 1024KB of flash!
[ INFO] [1626381922.886075837]: [cf1] DECK_CORE: 2 deck(s) found
[ INFO] [1626381922.908477997]: [cf1] DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
[ INFO] [1626381922.918345425]: [cf1] ZR2: Z-down sensor [OK]
[ INFO] [1626381922.941744172]: [cf1] PMW: Motion chip id: 0x49:0xB6
[ INFO] [1626381922.964189949]: [cf1] DECK_CORE: Calling INIT on driver bcMultiranger for deck 1
[ INFO] [1626381922.985593684]: [cf1] IMU: BMI088: Using I2C interface.
[ INFO] [1626381923.020959915]: [cf1] IMU: BMI088 Gyro connection [OK].
[ INFO] [1626381923.042203941]: [cf1] IMU: BMI088 Accel connection [OK]
[ INFO] [1626381923.064774649]: [cf1] IMU: BMP388 I2C connection [OK]
[ INFO] [1626381923.088126145]: [cf1] ESTIMATOR: Using Kalman (2) estimator
[ INFO] [1626381923.111552435]: [cf1] CONTROLLER: Using PID (1) controller
[ INFO] [1626381923.133093243]: [cf1] MTR-DRV: Using brushed motor driver
[ INFO] [1626381923.144797467]: [cf1] EEPROM: I2C connection [OK].
[ INFO] [1626381923.156634859]: [cf1] STORAGE: Storage check [OK].
[ INFO] [1626381923.179361514]: [cf1] IMU: BMI088 gyro self-test [OK]
[ INFO] [1626381923.189624328]: [cf1] DECK_CORE: Deck 0 test [OK].
[ INFO] [1626381923.201396366]: [cf1] MR: Init front sensor [OK]
[ INFO] [1626381923.213060262]: [cf1] MR: Init back sensor [OK]
[ INFO] [1626381923.224824236]: [cf1] MR: Init up sensor [OK]
[ INFO] [1626381923.236485462]: [cf1] MR: Init left sensor [OK]
[ INFO] [1626381923.248364951]: [cf1] MR: Init right sensor [OK]
[ INFO] [1626381923.258937577]: [cf1] DECK_CORE: Deck 1 test [OK].
[ INFO] [1626381923.283452316]: [cf1] STAB: Wait for sensor calibration...
[ INFO] [1626381923.294115121]: [cf1] SYS: Free heap: 16352 bytes
[ INFO] [1626381923.305776593]: [cf1] STAB: Ready to fly.
[ INFO] [1626381923.327660450]: [cf1] , resetting
[ INFO] [1626381923.348836432]: [cf1] Requesting parameters...
[ INFO] [1626381923.780238727]: Found variables in cache.
[ INFO] [1626381926.400499757]: [cf1] reqParamTOC: 3.644454 s (229 params)
[ INFO] [1626381926.400537933]: [cf1] Requesting logging variables...
[ INFO] [1626381926.429960570]: Found variables in cache.
[ INFO] [1626381926.430676266]: [cf1] reqLogTOC: 0.030175 s
[ INFO] [1626381926.495679669]: [cf1] logBlocks: 3.644454 s
[ INFO] [1626381926.495709940]: Requesting memories...
[ INFO] [1626381926.540126907]: Memories: 6
[ INFO] [1626381926.745026470]: [cf1] Ready. Elapsed: 3.988957 s
[ INFO] [1626381926.745071170]: CF run: 3.989095 s
update commander/enHighLevel to 1
update kalman/resetEstimation to 1
update locSrv/extPosStdDev to 0.001
update locSrv/extQuatStdDev to 0.05
update ring/effect to 16
update ring/headlightEnable to 0
update ring/solidBlue to 255
update ring/solidGreen to 0
update ring/solidRed to 0
update stabilizer/controller to 2
update stabilizer/estimator to 2
update ctrlMel/i_range_m_z to 1500
update ctrlMel/i_range_xy to 2
update ctrlMel/i_range_z to 2
update ctrlMel/kR_xy to 50000
update ctrlMel/kR_z to 60000
update ctrlMel/kd_omega_rp to 200
update ctrlMel/kd_xy to 1
update ctrlMel/kd_z to 1
update ctrlMel/ki_m_z to 500
update ctrlMel/ki_xy to 0.1
update ctrlMel/ki_z to 0.5
update ctrlMel/kp_xy to 2
update ctrlMel/kp_z to 3
update ctrlMel/kw_xy to 16000
update ctrlMel/kw_z to 12000
update ctrlMel/mass to 0.11
update ctrlMel/massThrust to 23000
[ INFO] [1626381926.756736557]: [cf1] Update parameters
[ INFO] [1626381927.210656635]: [cf1] CONTROLLER: Using Mellinger (2) controller
[ INFO] [1626381927.600737774]: Update params: 0.855633 s
[ INFO] [1626381927.601114506]: Started 1 threads

**Comment by tbaranoski** [comment]
I should also add that I do not see the drone visual on RVIZ.
![image](https://user-images.githubusercontent.com/84639789/125856456-8c1ee01e-133d-491e-8c24-6fb617d7ade1.png)

**Reply by whoenig** [reply]
It is normal that you don't see a visualization if you don't have a mocap system. You can, however, enable logging of the pose in the launch file and then visualize the onboard state estimate in rviz.

From your log everything seems to be working just fine. All you need now is to either get your joystick to work, or run another script in another terminal that actually sends commands do the drone. Your joystick seems to be detected. The "start" button should initiate takeoff (and the log should contain an entry for "Takeoff").
